File: projects/IST-IG-DD-JAVA/transforms-java/src/main/java/myproject/datasets/exposure/current/tr200_exposure.java
package myproject.datasets.exposure.current;

import com.palantir.transforms.lang.java.api.Compute;
import com.palantir.transforms.lang.java.api.Input;
import com.palantir.transforms.lang.java.api.Output;
import myproject.datasets.util.BP_DateUtils;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;

import java.util.Date;

public class tr200_exposure {
    public Dataset<Row> convert(Dataset<Row> df) {
        Date[] dates = BP_DateUtils.getStartEndDate();
        df = BP_DateUtils.filterLatest(df, dates[0], dates[1], "Current_COB_Date", "RunDateTime");
        return df;
    }

    @Compute
    @Output("/BP/IST-IG-DD/data/technical/exposure_java/current/titan/tr200_exposure")
    public Dataset<Row> myComputeFunction(
            @Input("/BP/IST-IG-DD/data/technical/exposure_java/typed/titan/tr200_exposure") Dataset<Row> df
    ){
        return convert(df);
    }
}

File: latest_integration/transforms-python/src/python/exposure/current/tr200_exposure.py
# ===== import: python function

# ===== import: palantir functions
from transforms.api import transform_df, Input, Output, configure

# ===== import: our functions
from python.util.bp_dateutils import filter_latest_runs
from python.util.bp_config import Config
from python.util.bp_constants import LOOK_BACK_PERIOD_CONFIG
from python.bp_flush_control import TR200_FLUSH, TR200_FLUSH_START_DATE, TR200_PROFILE


def convert(df):
    config = Config(look_back_period=LOOK_BACK_PERIOD_CONFIG,
                    flush=TR200_FLUSH,
                    flush_start_date=TR200_FLUSH_START_DATE)
    df = df.filter((df.Current_COB_Date >= config.start_date) & (df.Current_COB_Date <= config.end_date))
    df = filter_latest_runs(df, 'Current_COB_Date', 'RunDateTime')
    return df


@configure(profile=TR200_PROFILE)
@transform_df(
    Output("/BP/IST-IG-DD/data/technical/exposure/current/titan/tr200_exposure"),
    df=Input("/BP/IST-IG-DD/data/technical/exposure/typed/titan/tr200_exposure"),
)
def my_compute_function(df):
    return convert(df)
